src/Orleans/Serialization/ILBasedExceptionSerializer.cs

```diff
@@ -0,0 +1,231 @@
+﻿namespace Orleans.Serialization
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Reflection;
+
+    /// <summary>
+    /// Methods for serializing instances of <see cref="Exception"/> and its subclasses.
+    /// </summary>
+    internal class ILBasedExceptionSerializer
```

Comment:
That name is a bit long. I could call it `CilBasedExceptionSerializer`, but I think that's harder to read. This also entails renaming the existing IL* classes